apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/DmlReturningTests.cs

```diff
@@ -0,0 +1,186 @@
+            using (var connection = _fixture.GetConnection())
+            {
+                await RetryHelpers.ExecuteWithRetryAsync(async () =>
+                {
+                    string dml = $"UPDATE {_fixture.TableName} SET Value = OriginalValue + 1 WHERE UpdateMe AND Key=@key Then Return Value";
+                    using (var command = connection.CreateDmlCommand(dml))
+                    {
+                        command.Parameters.Add("key", SpannerDbType.String, key);
+                        Assert.Equal(2, await command.ExecuteNonQueryAsync());
+                    }
+                });
+            }
+            var actual = _fixture.FetchValues(key);
+            var expected = new Dictionary<int, int>
+            {
+                { 0, 0 }, // Not updated
+                { 1, 2 }, // Updated
+                { 2, 2 }, // Not updated
+                { 3, 3 }, // Not updated
+                { 4, 5 }  // Updated
+            };
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public async Task No_DMLReturn_ExecuteReader_Read()
+        {
+            if (_disableTest)
+                return;
+
+            string key = _fixture.CreateTestRows();
+            var actualValues = new List<int>();
+
+            using (var connection = _fixture.GetConnection())
+            {
+                await connection.OpenAsync();
+                RetryHelpers.ExecuteWithRetry(async () =>
+                {
+                    string dml = $"UPDATE {_fixture.TableName} SET Value = OriginalValue + 1 WHERE UpdateMe AND Key=@key";
+                    var command = connection.CreateDmlCommand(dml);
+                    using (var reader = await command.ExecuteReaderAsync())
+                    {
+                        if (await reader.ReadAsync())
+                        {
+                            // TODO: Yet to decide behaviour.
+                            //actualValues.Add(reader.GetFieldValue<int>(reader.GetOrdinal("Value")));
+                        }
+                    }
+                });
+            }
+            var expectedValues = new List<int>() { 2, 5 };
+            Assert.Equal(expectedValues, actualValues);
+        }
+
+        [Fact]
+        public async Task No_DMLReturn_ExecuteReader_No_Read()
+        {
+            if (_disableTest)
+                return;
+
+            string key = _fixture.CreateTestRows();
+            var actualValues = new List<int>();
+
+            using (var connection = _fixture.GetConnection())
+            {
+                await connection.OpenAsync();
+                RetryHelpers.ExecuteWithRetry(async () =>
+                {
+                    string dml = $"UPDATE {_fixture.TableName} SET Value = OriginalValue + 1 WHERE UpdateMe AND Key=@key";
+                    var command = connection.CreateDmlCommand(dml);
+                    var reader = await command.ExecuteReaderAsync(); // TODO: Yet to decide behaviour.
```

Comment:
Side note: We should complete the research, update the document with the desired behavior and get consensus this week.